### ISSUE TITLE
Username required for mechanism `SCRAM-SHA-1` MongoParseError: Username required for mechanism `SCRAM-SHA-1`

### ISSUE THREAD

**Issue by gustavopaes** [mongodb-js/mongodb-core#377]
At __mongodb-core >= 3.1.6__ (used __node-mongodb-native >= 3.1.7__) I am getting this message error:

    Username required for mechanism `SCRAM-SHA-1` MongoParseError: Username required for mechanism `SCRAM-SHA-1`

With the same connection object config:

```js
const connConfig = {
    useNewUrlParser: true,
    authMechanism: process.env.DATABASE_AUTH_MECHANISM,
    readPreference: ReadPreference.SECONDARY_PREFERRED,
    replicaSet: process.env.DATABASE_REPLICASET,
    socketTimeoutMS: Number(process.env.DATABASE_SOCKET_TIMEOUT || 900000)
};

if(process.env.DATABASE_USER) {
    connConfig.auth = {
        user: process.env.DATABASE_USER,
        password: process.env.DATABASE_PASS
    };
}

MongoClient.connect(`mongodb://${process.env.DATABASE_URI}/${process.env.DATABASE_NAME}`, connConfig);
```

Using __node-mongodb-native 3.1.6__ connection works.

**Comment by mbroadst** [comment]
Hi @gustavopaes, can you confirm that all values above indeed _have_ values? You're reading from `process.env` for most of them, so you could potentially have a `null`/`undefined` in there.

**Comment by mbroadst** [comment]
It looks like in refactoring to pass our auth spec tests we made an assumption that connection strings were the primary way to provide auth info, ignoring `auth` passed into `MongoClient` options. I've made this [ticket](https://jira.mongodb.org/browse/NODE-1767) to track the issue, and the fix is in #378

**Comment by mbroadst** [comment]
Fixed and released in v3.1.10, please let us know if you have any additional issues!
